src/libcore/ops.rs

```diff
@@ -591,26 +591,41 @@ not_impl! { bool usize u8 u16 u32 u64 isize i8 i16 i32 i64 }
 ///
 /// # Examples
 ///
-/// A trivial implementation of `BitAnd`. When `Foo & Foo` happens, it ends up
-/// calling `bitand`, and therefore, `main` prints `Bitwise And-ing!`.
+/// In this example, the `BitAnd` trait is implemented for a `BooleanVector`
+/// struct.
 ///
 /// ```
 /// use std::ops::BitAnd;
 ///
-/// struct Foo;
-///
-/// impl BitAnd for Foo {
-///     type Output = Foo;
-///
-///     fn bitand(self, _rhs: Foo) -> Foo {
-///         println!("Bitwise And-ing!");
-///         self
+/// #[derive(Debug)]
+/// struct BooleanVector {
+///     value: Vec<bool>,
+/// };
+///
+/// impl BitAnd for BooleanVector {
+///     type Output = Self;
+///
+///     fn bitand(self, rhs: Self) -> Self {
+///         BooleanVector {
+///             value: self.value
+///                        .iter()
+///                        .zip(rhs.value.iter())
+///                        .map(|(x, y)| *x && *y)
+///                        .collect(),
```

Comment:
I did (see modeline).

But caveat: I first indented `iter()`, and *then* invoked `rustfmt`. In a moment I'll throw up another version that `rustfmt` accepts.

![screen shot 2016-08-18 at 5 23 47 pm](https://cloud.githubusercontent.com/assets/6801001/17791350/b3ee3644-6568-11e6-877c-a7043bdb5060.png)